[Bug]: test decorator breaks with new pytest importlib mode

### Bug summary

pytest is failing on colletnig units.

### Code for reproduction

```python
I'm trying to package your module as an rpm package. So I'm using the typical build, install and test cycle used on building packages from non-root account.
- "setup.py build"
- "setup.py install --root </install/prefix>"
- "pytest --import-mode=importlib" with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>
```


### Actual outcome

```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ xvfb-run -a /usr/bin/pytest -ra --import-mode=importlib
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=2178151746
rootdir: /home/tkloczko/rpmbuild/BUILD/matplotlib-3.5.0, configfile: pytest.ini, testpaths: lib
plugins: shutil-1.7.0, virtualenv-1.7.0, mock-3.6.1, cov-2.12.1, anyio-3.3.4, forked-1.3.0, xdist-2.3.0, flaky-3.7.0, tornasync-0.6.0.post2, console-scripts-1.2.0, trio-0.7.0, timeout-2.0.1, asyncio-0.16.0, freezegun-0.4.2, flake8-1.0.7, pyfakefs-4.5.3, hypothesis-6.29.3, benchmark-3.4.1, profiling-1.7.0, datadir-1.3.1, regressions-2.2.0, randomly-3.8.0, rerunfailures-9.1.1, yagot-0.5.0, requests-mock-1.9.3, subtests-0.5.0, easy-server-0.8.0, django-4.5.2
collected 4599 items / 28 errors / 4571 selected

================================================================================== ERRORS ==================================================================================
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_artist.py ___________________________________________________________
lib/matplotlib/tests/test_artist.py:125: in <module>
    def test_clipping_zoom(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_artist'
____________________________________________________________ ERROR collecting lib/matplotlib/tests/test_axes.py ____________________________________________________________
lib/matplotlib/tests/test_axes.py:49: in <module>
    def test_label_loc_vertical(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_axes'
_______________________________________________________ ERROR collecting lib/matplotlib/tests/test_backend_cairo.py ________________________________________________________
lib/matplotlib/tests/test_backend_cairo.py:12: in <module>
    def test_patch_alpha_coloring(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_backend_cairo'
________________________________________________________ ERROR collecting lib/matplotlib/tests/test_backend_pdf.py _________________________________________________________
lib/matplotlib/tests/test_backend_pdf.py:317: in <module>
    def test_pdf_eps_savefig_when_color_is_none(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_backend_pdf'
________________________________________________________ ERROR collecting lib/matplotlib/tests/test_backend_pgf.py _________________________________________________________
lib/matplotlib/tests/test_backend_pgf.py:19: in <module>
    baseline_dir, result_dir = _image_directories(lambda: 'dummy func')
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_backend_pgf'
_________________________________________________________ ERROR collecting lib/matplotlib/tests/test_backend_ps.py _________________________________________________________
lib/matplotlib/tests/test_backend_ps.py:196: in <module>
    def test_text_clip(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_backend_ps'
________________________________________________________ ERROR collecting lib/matplotlib/tests/test_backend_svg.py _________________________________________________________
lib/matplotlib/tests/test_backend_svg.py:90: in <module>
    def test_rasterized(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_backend_svg'
__________________________________________________________ ERROR collecting lib/matplotlib/tests/test_category.py __________________________________________________________
lib/matplotlib/tests/test_category.py:287: in <module>
    def test_overriding_units_in_plot(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_category'
________________________________________________________ ERROR collecting lib/matplotlib/tests/test_collections.py _________________________________________________________
lib/matplotlib/tests/test_collections.py:309: in <module>
    def test_collection_log_datalim(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_collections'
__________________________________________________________ ERROR collecting lib/matplotlib/tests/test_colorbar.py __________________________________________________________
lib/matplotlib/tests/test_colorbar.py:764: in <module>
    def test_axes_handles_same_functions(fig_ref, fig_test):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_colorbar'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_colors.py ___________________________________________________________
lib/matplotlib/tests/test_colors.py:1464: in <module>
    def test_norm_update_figs(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_colors'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_figure.py ___________________________________________________________
lib/matplotlib/tests/test_figure.py:569: in <module>
    def test_add_artist(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_figure'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_image.py ____________________________________________________________
lib/matplotlib/tests/test_image.py:111: in <module>
    def test_imshow_antialiased(fig_test, fig_ref,
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_image'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_lines.py ____________________________________________________________
lib/matplotlib/tests/test_lines.py:137: in <module>
    def test_no_subslice_with_transform(fig_ref, fig_test):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_lines'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_marker.py ___________________________________________________________
lib/matplotlib/tests/test_marker.py:68: in <module>
    def test_poly_marker(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_marker'
__________________________________________________________ ERROR collecting lib/matplotlib/tests/test_mathtext.py __________________________________________________________
lib/matplotlib/tests/test_mathtext.py:310: in <module>
    def test_spaces(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_mathtext'
__________________________________________________________ ERROR collecting lib/matplotlib/tests/test_patches.py ___________________________________________________________
lib/matplotlib/tests/test_patches.py:82: in <module>
    def test_rotate_rect_draw(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_patches'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_polar.py ____________________________________________________________
lib/matplotlib/tests/test_polar.py:102: in <module>
    def test_polar_wrap(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_polar'
______________________________________________________ ERROR collecting lib/matplotlib/tests/test_preprocess_data.py _______________________________________________________
lib/matplotlib/tests/test_preprocess_data.py:266: in <module>
    class TestPlotTypes:
lib/matplotlib/tests/test_preprocess_data.py:272: in TestPlotTypes
    def test_dict_unpack(self, plotter, fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_preprocess_data'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_scale.py ____________________________________________________________
lib/matplotlib/tests/test_scale.py:17: in <module>
    def test_log_scales(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_scale'
_______________________________________________________ ERROR collecting lib/matplotlib/tests/test_simplification.py _______________________________________________________
lib/matplotlib/tests/test_simplification.py:250: in <module>
    def test_closed_path_nan_removal(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_simplification'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_spines.py ___________________________________________________________
lib/matplotlib/tests/test_spines.py:81: in <module>
    def test_spine_nonlinear_data_positions(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_spines'
__________________________________________________________ ERROR collecting lib/matplotlib/tests/test_testing.py ___________________________________________________________
lib/matplotlib/tests/test_testing.py:19: in <module>
    def test_parametrize_with_check_figure_equal(a, fig_ref, b, fig_test):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_testing'
____________________________________________________________ ERROR collecting lib/matplotlib/tests/test_text.py ____________________________________________________________
lib/matplotlib/tests/test_text.py:618: in <module>
    def test_annotation_units(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_text'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_units.py ____________________________________________________________
lib/matplotlib/tests/test_units.py:179: in <module>
    def test_subclass(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_units'
___________________________________________________________ ERROR collecting lib/matplotlib/tests/test_usetex.py ___________________________________________________________
lib/matplotlib/tests/test_usetex.py:42: in <module>
    def test_empty(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_usetex'
__________________________________________________________ ERROR collecting lib/matplotlib/tests/test_widgets.py ___________________________________________________________
lib/matplotlib/tests/test_widgets.py:1276: in <module>
    def test_rect_visibility(fig_test, fig_ref):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_widgets'
_________________________________________________________ ERROR collecting lib/mpl_toolkits/tests/test_mplot3d.py __________________________________________________________
lib/mpl_toolkits/tests/test_mplot3d.py:162: in <module>
    def test_contourf3d_extend(fig_test, fig_ref, extend, levels):
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:410: in decorator
    _, result_dir = _image_directories(func)
../../BUILDROOT/python-matplotlib-3.5.0-2.fc35.x86_64/usr/lib64/python3.8/site-packages/matplotlib/testing/decorators.py:482: in _image_directories
    module_path = Path(sys.modules[func.__module__].__file__)
E   KeyError: 'test_mplot3d'
========================================================================= short test summary info ==========================================================================
ERROR lib/matplotlib/tests/test_artist.py - KeyError: 'test_artist'
ERROR lib/matplotlib/tests/test_axes.py - KeyError: 'test_axes'
ERROR lib/matplotlib/tests/test_backend_cairo.py - KeyError: 'test_backend_cairo'
ERROR lib/matplotlib/tests/test_backend_pdf.py - KeyError: 'test_backend_pdf'
ERROR lib/matplotlib/tests/test_backend_pgf.py - KeyError: 'test_backend_pgf'
ERROR lib/matplotlib/tests/test_backend_ps.py - KeyError: 'test_backend_ps'
ERROR lib/matplotlib/tests/test_backend_svg.py - KeyError: 'test_backend_svg'
ERROR lib/matplotlib/tests/test_category.py - KeyError: 'test_category'
ERROR lib/matplotlib/tests/test_collections.py - KeyError: 'test_collections'
ERROR lib/matplotlib/tests/test_colorbar.py - KeyError: 'test_colorbar'
ERROR lib/matplotlib/tests/test_colors.py - KeyError: 'test_colors'
ERROR lib/matplotlib/tests/test_figure.py - KeyError: 'test_figure'
ERROR lib/matplotlib/tests/test_image.py - KeyError: 'test_image'
ERROR lib/matplotlib/tests/test_lines.py - KeyError: 'test_lines'
ERROR lib/matplotlib/tests/test_marker.py - KeyError: 'test_marker'
ERROR lib/matplotlib/tests/test_mathtext.py - KeyError: 'test_mathtext'
ERROR lib/matplotlib/tests/test_patches.py - KeyError: 'test_patches'
ERROR lib/matplotlib/tests/test_polar.py - KeyError: 'test_polar'
ERROR lib/matplotlib/tests/test_preprocess_data.py - KeyError: 'test_preprocess_data'
ERROR lib/matplotlib/tests/test_scale.py - KeyError: 'test_scale'
ERROR lib/matplotlib/tests/test_simplification.py - KeyError: 'test_simplification'
ERROR lib/matplotlib/tests/test_spines.py - KeyError: 'test_spines'
ERROR lib/matplotlib/tests/test_testing.py - KeyError: 'test_testing'
ERROR lib/matplotlib/tests/test_text.py - KeyError: 'test_text'
ERROR lib/matplotlib/tests/test_units.py - KeyError: 'test_units'
ERROR lib/matplotlib/tests/test_usetex.py - KeyError: 'test_usetex'
ERROR lib/matplotlib/tests/test_widgets.py - KeyError: 'test_widgets'
ERROR lib/mpl_toolkits/tests/test_mplot3d.py - KeyError: 'test_mplot3d'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 28 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=========================================================================== 28 errors in 19.45s ============================================================================
```


### Expected outcome

pytets should be able at least collect units and start testing.

### Additional information

_No response_

### Operating system

Linux/x86_64

### Matplotlib Version

3.5.0

### Matplotlib Backend

_No response_

### Python version

3.8.12

### Jupyter version

_No response_

### Installation

from source (.tar.gz)

--- a/lib/matplotlib/testing/decorators.py
+++ b/lib/matplotlib/testing/decorators.py
@@ -502,7 +502,7 @@ def _image_directories(func):
     ``$(pwd)/result_images/test_baz``.  The result directory is created if it
     doesn't exist.
     """
-    module_path = Path(sys.modules[func.__module__].__file__)
+    module_path = Path(inspect.getfile(func))
     baseline_dir = module_path.parent / "baseline_images" / module_path.stem
     result_dir = Path().resolve() / "result_images" / module_path.stem
     result_dir.mkdir(parents=True, exist_ok=True)

